### PR TITLE
[ntuple] fix vector read from columns

### PR DESCRIPTION
**Pull request by lesimon74**
Fix implementation for the case where a std::string goes over the page limit. A small test is also added to test it.

### CODE REVIEW

**Comment by pcanal** [comment]
@phsft-bot build with flags -DCTEST_TEST_EXCLUDE_NONE=On

File: tree/ntuple/v7/test/ntuple.cxx [hunk 1]
```diff
@@ -787,3 +787,25 @@ TEST(RNTuple, Descriptor)
...
+{
+   const std::string_view ntupleName = "rs";
+   FileRaii fileGuard("test_ntuple_readstring.root");
+   {
+      auto model = RNTupleModel::Create();
+      auto st = model->MakeField<std::string>("st");
+      auto ntuple = RNTupleWriter::Recreate(std::move(model), ntupleName, fileGuard.GetPath());
+
+      for (int i = 0; i < 25000; ++i) {
+      *st = "foooooo";
```
Comment (jblomer) [inline tree/ntuple/v7/test/ntuple.cxx]: The indentation seems a bit off

File: tree/ntuple/v7/test/ntuple.cxx [hunk 2]
```diff
@@ -787,3 +787,25 @@ TEST(RNTuple, Descriptor)
...
+      ntuple->Fill();
+      }
+   }
+
+   auto ntuple = RNTupleReader::Open(ntupleName, fileGuard.GetPath());
+   auto viewSt = ntuple->GetView<std::string>("st");
+   // assumes NElements per Page is 10000
```
Comment (jblomer) [inline tree/ntuple/v7/test/ntuple.cxx]: You can ask the descriptor whether the assumption holds (multiple pages for the column...)

**Comment by lesimon74** [comment]
@phsft-bot build with flags -DCTEST_TEST_EXCLUDE_NONE=On

File: tree/ntuple/v7/test/ntuple.cxx [hunk 3]
```diff
@@ -787,3 +787,29 @@ TEST(RNTuple, Descriptor)
...
+      }
+   }
+
+   auto ntuple = RNTupleReader::Open(ntupleName, fileGuard.GetPath());
+   auto viewSt = ntuple->GetView<std::string>("st");
+   if (ntuple->GetDescriptor().GetClusterDescriptor(0).GetPageRange(1).fPageInfos.size() < 2) {
+      FAIL(); // This means all entries are inside the same page and numEntries should be increased.
+   }
+   int nElementsPerPage = ntuple->GetDescriptor().GetClusterDescriptor(0).GetPageRange(1).fPageInfos.at(1).fNElements;
+   EXPECT_EQ(viewSt(nElementsPerPage/7), "foooooo");
```
Comment (jblomer) [inline tree/ntuple/v7/test/ntuple.cxx]: I forgot, in gtest, when testing for equality, the arguments go the other way round: `"foooooo", viewSt...`

**Pull request merged**